bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -1230,7 +1232,16 @@ void checkAllLedgers() throws BKException, IOException, InterruptedException, Ke
                     return;
                 }
 
+                try {
+                    semaphore.acquire();
```

Comment:
I'd consider using tryAcquire() with some reasonable timeout and either just treating the timeout similarly to `InterruptedException` (return) or retry in the loop. Plus add logging on the timeout.
Otherwise one will have to deal with thread dumps to understand why auditor is not progressing when it is stuck on that semaphore in case of some obscure bug in zk client or server.